dask/bag/core.py

```diff
@@ -68,12 +69,34 @@ def lazify(dsk):
     return valmap(lazify_task, dsk)
 
 
+def inline_singleton_lists(dsk):
```

Comment:
If `inline_functions` was made to optionally take a mapping of `{function: n_dependents}` instead of just a set of functions, then this could be replaced with `inline_functions(dsk, fast_functions={list: 1})`. I see no problem with adding another, more specific optimization function - just a thought. Is there a reason you don't do a full `fuse` here (would have same effect, but with even more inlining)?